ipaserver/install/dsinstance.py

```diff
@@ -1041,23 +1070,42 @@ def uninstall(self):
 
         try:
             self.fstore.restore_file(paths.LIMITS_CONF)
+        except ValueError as error:
+            logger.debug("%s: %s", paths.LIMITS_CONF, error)
+
+        try:
             self.fstore.restore_file(paths.SYSCONFIG_DIRSRV)
         except ValueError as error:
-            logger.debug("%s", error)
+            logger.debug("%s: %s", paths.SYSCONFIG_DIRSRV, error)
 
         # disabled during IPA installation
         if enabled:
+            logger.debug("Re-enabling instance of Directory Server")
             self.enable()
 
         serverid = self.restore_state("serverid")
         if serverid is not None:
+            # What if this fails? Then what?
             self.stop_tracking_certificates(serverid)
             logger.debug("Removing DS instance %s", serverid)
             try:
                 remove_ds_instance(serverid)
             except ipautil.CalledProcessError:
                 logger.error("Failed to remove DS instance. You may "
                              "need to remove instance data manually")
+            try:
+                installutils.remove_keytab(paths.DS_KEYTAB)
+            except ipautil.CalledProcessError:
+                logger.error("Failed to remove DS keytab. You may "
```

Comment:
``installutils.remove_keytab()`` and ``installutils.remove_ccache()`` already handle exceptions and perform logging. You don't handle additional exceptions here.